pkg/apply/run.go

```diff
@@ -162,12 +163,12 @@ func (r *ClusterArgs) SetClusterRunArgs(imageList []string, args *RunArgs) error
 		masters := stringsutil.SplitRemoveEmpty(args.Cluster.Masters, ",")
 		nodes := stringsutil.SplitRemoveEmpty(args.Cluster.Nodes, ",")
 		r.hosts = []v2.Host{}
-		if len(masters) != 0 {
-			r.setHostWithIpsPort(masters, []string{v2.MASTER, string(v2.AMD64)})
-		}
-		if len(nodes) != 0 {
-			r.setHostWithIpsPort(nodes, []string{v2.NODE, string(v2.AMD64)})
-		}
+
+		clusterSSH := r.cluster.GetSSH()
+		sshClient := ssh.NewSSHClient(&clusterSSH, true)
+
+		r.setHostWithIpsPort(masters, []string{v2.MASTER, GetHostArch(sshClient, masters[0])})
```

Comment:
If there is something wrong with different nodes and different architectures, I feel traversed, right?

Reply:
Sorry, I do not quite understand what you mean. Does "different nodes and different architectures" means that we have masters with different arch or nodes with different arch? I think we currently only support masters/nodes with the same arch.

Reply:
> Sorry, I do not quite understand what you mean. Does "different nodes and different architectures" means that we have masters with different arch or nodes with different arch? I think we currently only support masters/nodes with the same arch.

At present, it is like this, but in the future, different nodes should be satisfied with different architectures.

Reply:
We can handle the mixed architecture case in a single new feature